plugins/wp-graphql-headless-webhooks/src/Events/WebhookEventManager.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace WPGraphQL\Webhooks\Events;
+
+use WPGraphQL\Webhooks\Events\Interfaces\EventManager;
+use WPGraphQL\Webhooks\Repository\Interfaces\WebhookRepositoryInterface;
+use WPGraphQL\Webhooks\Handlers\Interfaces\Handler;
+
+/**
+ * Webhook Event Manager
+ *
+ * Manages WordPress events and triggers matching webhooks.
+ */
+class WebhookEventManager implements EventManager {
+
+	private WebhookRepositoryInterface $repository;
+	private Handler $handler;
+
+	/**
+	 * Constructor
+	 *
+	 * @param WebhookRepositoryInterface $repository
+	 * @param Handler     $sender
+	 */
+	public function __construct( WebhookRepositoryInterface $repository, $handler ) {
+		$this->repository = $repository;
+		$this->handler = $handler;
+	}
+
+	/**
+	 * Register specific WordPress event hooks.
+	 */
+	public function register_hooks(): void {
+		add_action( 'transition_post_status', [ $this, 'on_transition_post_status' ], 10, 3 );
+		add_action( 'post_updated', [ $this, 'on_post_updated' ], 10, 3 );
+		add_action( 'deleted_post', [ $this, 'on_deleted_post' ], 10, 2 );
+		add_action( 'added_post_meta', [ $this, 'on_post_meta_change' ], 10, 4 );
+		add_action( 'created_term', [ $this, 'on_term_created' ], 10, 3 );
+		add_action( 'set_object_terms', [ $this, 'on_term_assigned' ], 10, 6 );
+		add_action( 'delete_term_relationships', [ $this, 'on_term_unassigned' ], 10, 3 );
+		add_action( 'delete_term', [ $this, 'on_term_deleted' ], 10, 4 );
+		add_action( 'added_term_meta', [ $this, 'on_term_meta_change' ], 10, 4 );
+		add_action( 'user_register', [ $this, 'on_user_created' ], 10, 1 );
+		add_action( 'deleted_user', [ $this, 'on_user_deleted' ], 10, 2 );
+		add_action( 'add_attachment', [ $this, 'on_media_uploaded' ], 10, 1 );
+		add_action( 'edit_attachment', [ $this, 'on_media_updated' ], 10, 1 );
+		add_action( 'delete_attachment', [ $this, 'on_media_deleted' ], 10, 1 );
+		add_action( 'wp_insert_comment', [ $this, 'on_comment_inserted' ], 10, 2 );
+		add_action( 'transition_comment_status', [ $this, 'on_comment_status' ], 10, 3 );
```

Comment:
@theodesp @josephfusco 

One thing which might be nice is to allow developers specify their own hooks (whether custom or core) that could also run the trigger_webhook too.

Reply:
Why aren't we using smart cache to handle all this? I thought that was the whole point of this plugin?

Reply:
We can't rely on the Smart Cache plugin for this at the moment. The webhook system needs to be decoupled from caching strategies because webhooks should fire reliably regardless of cache state, and Smart Cache currently doesn't cover all the event hooks we need or we might need.

I think it’s better that we manage our own hooks for now and if/when Smart Cache adds proper integration points, or if we decide to extend it ourselves, we could refactor this to optionally skip registering these actions in favor of Smart Cache triggers. Happy to open a PR for that integration down the line.